libeos-payg/efi.c

```diff
@@ -669,6 +711,71 @@ eospayg_efi_var_read (const char  *name,
   return ret;
 }
 
+/* eospayg_efi_var_read_fullname_boolean:
+ * @name: Full name of variable
+ * @error: return location for an error, or %NULL
+ *
+ * Read the contents of an EFI variable as a boolean with its full name.
+ *
+ * It only checks the first byte of the EFI variable's content.
+ *
+ * Returns: (transfer full): A boolean of the variable content, or %NULL on
+ *          error
+ */
+gboolean
+eospayg_efi_var_read_fullname_boolean (const char  *name,
+                                       GError     **error)
+{
+  g_return_val_if_fail (efi != NULL, FALSE);
+  g_return_val_if_fail (name != NULL, FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  g_autofree unsigned char *content = NULL;
+  int size;
+
+  content = eospayg_efi_var_read_fullname (name, -1, &size, error);
+  if (!content || size < 1)
```

Comment:
On this code path if `size` is less than 1 you are returning `FALSE` but not setting `error`. I guess it's also the case that `error` is not set when `content[0]` is zero...

Typically the convention is that a false-y return value should always set the error. In practice it doesn't matter because the only caller is inside this source file and it handles that case directly. Perhaps make this function `static`.

Reply:
Actually if you check the implementation of `efivarfs_read` it is impossible for `size` to be zero but `content` to be non-NULL. Theoretically `test_read` can return zero `size` but non-NULL `content` but only if someone incorrectly writes an empty variable. According to other comments in this code, experimentation, the corresponding Windows API, and the [SetVariable](https://uefi.org/specs/UEFI/2.10/08_Services_Runtime_Services.html#setvariable) UEFI function, the variable data cannot be zero bytes long.